pkg/extension/extension.go

```diff
@@ -0,0 +1,22 @@
+package extension
+
+import (
+	"context"
+
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/release"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+type Extension interface{}
```

Comment:
Since the Extension interface is completely unrelated to the specific extension types, I'm a little worried that compiler will not complain and an extension will simply be silently ignored in case someone has a typo in the method name.

Not sure if there is a good solution, but perhaps an example where you do something like `var _ PreReconciliationExtension = &myThing{}` idiom?

Reply:
I see the problem as well. I tend to believe that the advantages of this approach outweigh its disadvantages, though. Definitely a :+1: for the idea with the example.